doc/crypto/api/ops/encap.rst

```diff
@@ -0,0 +1,326 @@
+.. function:: psa_encapsulate_key
+
+    .. summary::
+        Generate a new key pair and a use that to encapsulate a new symmetric key, emitting it both as a key object and an encapsulation to send to a counter party along with the public key from the ephemeral key pair.
+
+    .. param:: const psa_key_id_t * counterparty_key
+        The identifier for the public key of the peer. You must have previously imported this key using `psa_import_key()`, and specified the key attributes for the public key type corresponding to the type required for the encapsulation, and the usage usage `PSA_KEY_USAGE_ENCAPSULATE_KEY`.
+
+    .. param:: psa_key_id_t * key_pair
+        On success an identifier for the new key pair, which was used to perform the encapsulation.  
+        The key will be the key type specified in the atributes of the counterparty key. The private key will be given the usage `PSA_KEY_USAGE_ENCAPSULATE_KEY`. The public key only will be given the usage `PSA_KEY_USAGE_EXPORT`.  
+        `PSA_KEY_ID_NULL` on failure
+
+    .. param:: uint8_t * ephemeral_public_key
+        Buffer where the ephemeral public key key is to be written, ready to be sent to the counterparty.
+        
+    .. param:: size_t ephemeral_public_key_size
+        Size of the ``ephemeral_public_key`` buffer in bytes.
+        This must be at least :code:`PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(alg)`.
+        A buffer of at least :code:`PSA_EXPORT_PUBLIC_KEY_MAX_OUTPUT_SIZE`. is guaranteed not to fail due to buffer size for any supported encapsulation algorithm.
+
+    .. param:: psa_algorithm_t alg
+        The ful encapsulation algorithm to use: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_ENCAPSULATION(alg)` is true and :code:`PSA_ALG_IS_RAW_ENCAPSULATION(alg)` is false .
+
+    .. param:: const psa_key_attributes_t * attributes
+        The attributes for the new symmetric key.
+        This function uses the attributes as follows:
+
+        *   The key type is required. It cannot be an asymmetric public key.
+        *   The key size is required. It must be a valid size for the key type.
+        *   The key permitted-algorithm policy is required for keys that will be used for a cryptographic operation, see :secref:`permitted-algorithms`.
+        *   The key usage flags define what operations are permitted with the key, see :secref:`key-usage-flags`.
+        *   The key lifetime and identifier are required for a persistent key.
+
+        .. note::
+            This is an input parameter: it is not updated with the final key attributes. The final attributes of the new key can be queried by calling `psa_get_key_attributes()` with the key's identifier.
+        
+
+    .. param:: psa_key_id_t * output_key
+        On success, an identifier for the newly created key. `PSA_KEY_ID_NULL` on failure.
+        
+    .. param:: uint8_t * encapsulation
+        Buffer where the encapsulated key is to be written, ready to be sent to the counterparty.

```

Comment:
In general, the encapsulation data might contain a key (as for ECIES), but might contain entirely different data (as for ML-KEM).